### PR TITLE
enable codegen for java automation, and adjust sdk generation script

### PR DESCRIPTION
**Pull request by zzvswxy**
# Description

enable codegen for java automation, and adjust sdk generation script.
  codegen_to_sdk_config.json: codegen config file and must at root dir.
  eng/mgmt/automation/sdk_generate.py&sdk_init.sh: changed from generate.py&init.sh for codegen
  eng/mgmt/automation/generate_utils.py: common funcs for sdk_generate.py and generate.py

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by weidongxu-microsoft** [comment]
Please describe what are the changes.

**Comment by zzvswxy** [comment]
> Please describe what are the changes.

I update the 'Description' for what&why change the files. Thx

File: codegen_to_sdk_config.json
Comment (weshaggard) [inline codegen_to_sdk_config.json:1]: Lets please put this file somewhere under the eng folder instead of in the root of the repo.

**Comment by weshaggard** [comment]
@weidongxu-microsoft @zzvswxy why do we need another swagger configuration here? Can we not reuse the existing automation?

**Comment by weidongxu-microsoft** [comment]
@weshaggard 

@zzvswxy would give the background (I am aware of the idea but not the detail). (I thought team might have discussion on EngSys sync meeting)

Offline discussion is that the PR is on hold for Crystal to join as well. And concur that file should not be put to project root unless really necessary.

**Comment by chunyu3** [comment]
> 

@weshaggard We are developing SDK Generation which will be used in automation pipleline, shif-left and so on. The SDK Generation  will read codegen_to_sdk_config.json to get scripts to run and the script running options, the similar purpose of 'swagger_to_sdk_config.json'. We design the config and the way to run script the similar way to reuse the swagger sdk automation scripts as much as possible. 

For the location of the codegen_to_sdk_config.json, we will move it to the eng folder. We will do it together for all SDKs and the backend task execution engine. We have filed an issue for this. https://github.com/Azure/azure-sdk-for-go/issues/16863

**Comment by weshaggard** [comment]
@chunyu3 I'm still a little unclear on why we need a parallel set of configuration files and scripts from the existing swagger_to_sdk config and scripts. I would expect the new sdk automation would use the same scripts, possibly with some updates. This will be a good topic for this week sync meeting I will add it to the agenda.

**Comment by chunyu3** [comment]
> @chunyu3 I'm still a little unclear on why we need a parallel set of configuration files and scripts from the existing swagger_to_sdk config and scripts. I would expect the new sdk automation would use the same scripts, possibly with some updates. This will be a good topic for this week sync meeting I will add it to the agenda.

Just quick answer: The codegen_to_sdk_config is extend the swagger_to_sdk_config to add more task config (such as mocktest, livetest) and will update the task execution options and input/output. And we will reuse the scripts as more as possible. In this PR, we just refact the script to retrieve out the common scripts as common lib which will be used by both swagger-pr-pipeline and sdk-generation-automation/shift-left.  We will discuss detail in sync meeting.

File: eng/mgmt/automation/sdk_generate.py
Comment (weidongxu-microsoft) [inline eng/mgmt/automation/sdk_generate.py:259]: nit, refresh main, consider use `stream=sys.stdout,` here (if you do not have reason to log to stderr).

File: eng/mgmt/automation/sdk_generate.py
Comment (weidongxu-microsoft) [inline eng/mgmt/automation/sdk_generate.py:59]: nit, not sure if you need these args in your automation script. Anyway, your script, yours to maintain.

**Review by weidongxu-microsoft — APPROVED** [review]
There is some modification on `generate` and `compare_package` but they are moved across files so I cannot able to read the change.

Let me know you've reviewed the comment and make all changes you want to make. I will need to override check-enforcer after your last commit for merge.

File: eng/mgmt/automation/sdk_generate.py
Comment (zzvswxy) [reply eng/mgmt/automation/sdk_generate.py:59]: > nit, not sure if you need these args in your automation script. Anyway, your script, yours to maintain.

update with your changes is generate.py

File: eng/mgmt/automation/sdk_generate.py
Comment (zzvswxy) [reply eng/mgmt/automation/sdk_generate.py:259]: update the file with  your changes is generate.py, thx

File: eng/mgmt/automation/sdk_generate.py
Comment (zzvswxy) [reply eng/mgmt/automation/sdk_generate.py:59]: add it first for being compatible with existing functions and will discuss whether delete or change it

**Pull request merged**
